frontend/lib/evictionfree/declaration-builder/routes.tsx

```diff
@@ -272,6 +276,43 @@ export const getEvictionFreeDeclarationBuilderProgressRoutesProps = (): Progress
   };
 };
 
-export const EvictionFreeDeclarationBuilderRoutes = buildProgressRoutesComponent(
+/**
+ * Full EvictionFree Routes component for when the tool is active and allowing users to
+ * generate and send declaration forms.
+ */
+const FullEvictionFreeDeclarationBuilderRoutes = buildProgressRoutesComponent(
   getEvictionFreeDeclarationBuilderProgressRoutesProps
 );
+
+/**
+ * Modified EvictionFree Routes component for when the tool is suspended and blocking users from
+ * generating and sending declaration forms. This modified component takes note of the current
+ * location of the user and will redirect them back to the homepage unless they are on one of
+ * the "Excluded Routes" that we still want to keep active even when the tool is suspended.
+ */
+const SuspendedEvictionFreeDeclarationBuilderRoutes = () => {
+  const location = useLocation();
```

Comment:
Seems like this might be the place codeclimate is complaining about.
You would probably mock out `useLocation()` (https://stackoverflow.com/questions/59949052/how-do-you-mock-uselocation-pathname-using-shallow-test-enzyme-reactjs/60173736), and just check it returns `FullEvictionFree...` or `EvictionFreeRedirect` for the two cases we care about.

Reply:
going to file a "Shortcut" card for this after merge, thanks for flagging